app/addons/documents/changes/components.react.jsx

```diff
@@ -21,168 +21,174 @@ import ReactComponents from "../../components/react-components.react";
 import ReactCSSTransitionGroup from "react-addons-css-transition-group";
 import "../../../../assets/js/plugins/prettify";
 
-var changesStore = Stores.changesStore;
-
+var store = Stores.changesStore;
 var BadgeList = ReactComponents.BadgeList;
 
-// the top-level component for the Changes Filter section. Handles hiding/showing of the filters form
-var ChangesHeaderController = React.createClass({
-  getInitialState: function () {
-    return this.getStoreState();
-  },
 
-  getStoreState: function () {
+class ChangesController extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = this.getStoreState();
+  }
+
+  getStoreState () {
     return {
-      showTabContent: changesStore.isTabVisible()
+      changes: store.getChanges(),
+      loaded: store.isLoaded(),
+      databaseName: store.getDatabaseName(),
+      isShowingSubset: store.isShowingSubset()
     };
-  },
+  }
 
-  onChange: function () {
+  onChange () {
     this.setState(this.getStoreState());
-  },
-
-  componentDidMount: function () {
-    changesStore.on('change', this.onChange, this);
-  },
+  }
 
-  componentWillUnmount: function () {
-    changesStore.off('change', this.onChange);
-  },
+  componentDidMount () {
+    store.on('change', this.onChange, this);
```

Comment:
Actually not in this instance. React doesn't handle all the binding for us, so we need to explicitly bind it so that onChange can reference this.getStoreState().